### PR TITLE
added config setting to prune warpdb

### PR DESCRIPTION
**Pull request by morrisettjohn**
## Why this should be merged
Adds a config that lets validators prune their warpdb messages on initialization
## How this works
Clears via clearprefix

## How this was tested
Via a unit test included in test

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -38,10 +38,10 @@ require (
...
 	github.com/tyler-smith/go-bip39 v1.1.0
 	github.com/urfave/cli/v2 v2.17.2-0.20221006022127-8f469abc00aa
-	golang.org/x/crypto v0.1.0
-	golang.org/x/sync v0.1.0
-	golang.org/x/sys v0.8.0
-	golang.org/x/text v0.7.0
+	golang.org/x/crypto v0.10.0
+	golang.org/x/sync v0.3.0
+	golang.org/x/sys v0.9.0
+	golang.org/x/text v0.10.0
```
Comment (aaronbuchwald) [inline go.mod:44]: can we reset the changes to `go.mod` and `go.sum` ? 

We usually cautiously update dependencies and try to avoid coupling it to unrelated changes

File: plugin/evm/vm.go [hunk 2]
```diff
@@ -272,6 +272,13 @@ func (vm *VM) Initialize(
 	vm.metadataDB = prefixdb.New(metadataPrefix, vm.db)
 	vm.warpDB = prefixdb.New(warpPrefix, vm.db)
 
+	if vm.config.PruneWarpDB {
+		err := database.ClearPrefix(vm.db, vm.db, warpPrefix)
+		if err != nil {
+			return fmt.Errorf("failed to prune warpDb")
+		}
+	}
```
Comment (aaronbuchwald) [inline plugin/evm/vm.go:280]: Will this clear the correct data passing in `warpPrefix` to `database.ClearPrefix` to clear the database created by `prefixdb.New(warpPrefix, ...)` ?

File: plugin/evm/vm_test.go [hunk 3]
```diff
@@ -3204,3 +3206,83 @@ func TestSignatureRequestsToVM(t *testing.T) {
 	err = vm.Network.AppRequest(context.Background(), ids.GenerateTestNodeID(), 1, time.Now().Add(60*time.Second), requestBytes)
 	require.NoError(t, err)
 }
+
+// Test that the PruneWarpDB config flag applies as expected
+func TestPruneWarpDB(t *testing.T) {
+	// Because the chain is being reinitialized, metrics need to be disabled and upgrades skipped to allow initialization
+	t.Skip("no skipable upgrades")
```
Comment (aaronbuchwald) [inline plugin/evm/vm_test.go:3213]: Should we remove `Skip` ? It seems that this test fails for me if I remove it atm

File: go.mod (hunk 1 above)
Comment (morrisettjohn) [reply go.mod:44]: Sure thing, I wasn't entirely sure how those were updated anyways

File: plugin/evm/vm.go (hunk 2 above)
Comment (morrisettjohn) [reply plugin/evm/vm.go:280]: Sounds good to me!

File: plugin/evm/vm_test.go (hunk 3 above)
Comment (morrisettjohn) [reply plugin/evm/vm_test.go:3213]: Probably better to remove it, also since the unit test is being moved to the backend this entire unit test can probably be removed anyways

**Review by aaronbuchwald — APPROVED** [review]

File: plugin/evm/config.go [hunk 4]
```diff
@@ -122,6 +123,7 @@ type Config struct {
 	AllowMissingTries               bool    `json:"allow-missing-tries"`                // If enabled, warnings preventing an incomplete trie index are suppressed
 	PopulateMissingTries            *uint64 `json:"populate-missing-tries,omitempty"`   // Sets the starting point for re-populating missing tries. Disables re-generation if nil.
 	PopulateMissingTriesParallelism int     `json:"populate-missing-tries-parallelism"` // Number of concurrent readers to use when re-populating missing tries on startup.
+	PruneWarpDB                     bool    `json:"prune-warp-db-enabled"`              // Determines if the warpDb should be cleared when re-initialzing subnet-evm
```
Comment (anusha-ctrl) [inline plugin/evm/config.go:127]: nit: re-initialzing -> reinitializing

File: warp/backend_test.go [hunk 5]
```diff
@@ -21,6 +21,42 @@ var (
...
+
+	// ensure all messages have been deleted
+	for _, messageID := range messageIDs {
+		_, err := backend.GetSignature(messageID)
+		require.ErrorContains(t, err, "failed to get warp message")
+	}
+}
+
 func TestAddAndGetValidMessage(t *testing.T) {
 	db := memdb.New()
```
Comment (anusha-ctrl) [inline warp/backend_test.go:62]: do we need to add a e2e test in in `vm_test.go`?

**Review by darioush — APPROVED** [review]

File: plugin/evm/config.go [hunk 6]
```diff
@@ -122,6 +123,7 @@ type Config struct {
...
```
Comment (ceyonur) [inline plugin/evm/config.go:126]: Should we add add a validation for the case `PruneWarpDB` was enabled but `Pruning` was not? If these are not related, can we rename it to something like `warp-db-cleanup-enabled` etc? lIMHO one can confuse this with `pruning` and try to enable both of them.

**Review by ceyonur — COMMENTED** [review]
Overall LGTM. just a concern with the config name

File: plugin/evm/config.go (hunk 6 above)
Comment (aaronbuchwald) [reply plugin/evm/config.go:126]: These are two different types of pruning.

I think pruning is the right term here tbh

Comment (aaronbuchwald) [inline plugin/evm/config.go:126]:
```suggestion
	PruneWarpDB                     bool    `json:"prune-warp-db-enabled"`              // Determines if the warpDB should be cleared on startup
```

File: plugin/evm/config.go (hunk 4 above)
Comment (aaronbuchwald) [reply plugin/evm/config.go:127]: good catch updated to cleared on startup

**Review by aaronbuchwald — APPROVED** [review]
